apps/frontend/src/modules/surveys/components/studio/create/editor-panel/components/EditorTextInput.tsx

```diff
@@ -21,89 +25,137 @@ export const EditorTextInput = ({
     className,
     showCharacterCount = true,
     maxCharacterCount = 500,
-    placeholder,
     defaultValue,
+    options,
     onChange,
-    ...props
 }: EditorTextProps) => {
-    const [atBtnClicked, setAtBtnClicked] = useState(false);
-    const [atIndex, setAtIndex] = useState<number | null>(null);
-    const [selectedOption, setSelectedOption] = useState<Option | null>(null);
-    const [inputValue, setInputValue] = useState(defaultValue || "");
+    const [textContent, setTextContent] = useState(decode(defaultValue ?? ""));
+
+    const [displayFallbackField, setDisplayFallbackField] = useState(false);
+
+    const modules: StringMap = useMemo(() => {
+        return {
+            toolbar: false,
+            syntax: false,
+            mention: {
+                allowedChars: /^[A-Za-z\sÅÄÖåäö]*$/,
+                allowInlineMentionChar: true,
+                isolateCharacter: true,
+                mentionDenotationChars: ["@"],
+                showDenotationChar: false,
+                spaceAfterInsert: true,
+                positioningStrategy: "absolute",
+                source: function (searchTerm, renderList, mentionChar) {
+                    if (searchTerm.startsWith(" ")) {
+                        renderList([]);
+                        return;
+                    }
+                    renderList(options);
+                },
+                onSelect: function (item, insertItem) {
+                    const newItem = item;
+                    const details = newItem.id.split(" ");
+                    insertItem({ ...newItem, value: item.value, id: details[0], type: details[1] });
+                },
+            },
+        };
+    }, []);
+
+    function resolveQuestionIndex(questionId: string): string {
+        const option = options?.find((o) => {
+            const optionId = o.id.split(" ")[0];
+            return questionId === optionId;
+        });
+
+        return option?.value as string;
+    }
+
+    function encode(textContent: string): string {
```

Comment:
@afolasope Can you move these functions to a helper file? They don't have any side effect.